### PR TITLE
add function for bool in math/rand

### LINKED ISSUE #23804: Add function for random bool in math/rand
I want to be able to call `rand.Bool()` and receive a pseudo-random bool when I import `math/random`

### PR DESCRIPTION
**Pull request by kirinrastogi**
Closes #23804

I wasn't sure how to test this.

### CODE REVIEW

**Comment by kirinrastogi** [comment]
I signed it!

File: src/math/rand/rand.go [hunk 1]
```diff
@@ -210,6 +210,11 @@ again:
 	return f
 }
 
+// Bool returns a pseudo-random bool in [true, false].
```
Comment (mewmew) [inline src/math/rand/rand.go]: While `true` and `false` is not specified as as `1` and `0` (respectively) in Go, given the widespread use of that definition in other languages, the range should be swapped, as follows.

```diff
-[true, false].
+[false, true].
```

**Comment by WissCoast** [comment]
@kirinrastogi any updates?

File: src/math/rand/rand.go [hunk 2]
```diff
@@ -210,6 +210,11 @@ again:
...
+func (r *Rand) Bool() bool {
+	return r.Int63()%2 == 0
```
Comment (bbuck) [inline src/math/rand/rand.go:215]: Would it not be more efficient to do something like this?

```go
	return r.Int63() & 1 == 0
```

**Pull request closed without merging**